client/python/gradio_client/client.py

```diff
@@ -235,7 +235,8 @@ def duplicate(
         current_hardware = (
             current_info.hardware or huggingface_hub.SpaceHardware.CPU_BASIC
         )
-        hardware = hardware or original_info.hardware
+        if hardware is None and original_info.hardware is not None:
```

Comment:
Fixed a typing error here

Reply:
What's the type error? I thought these are equivalent.

<img width="224" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/7ab03dfe-f9cd-4ccf-8858-1603b9e9c013">

Reply:
Ok so it was the fact that pyright didn't like `hardware` getting reassigned to have a different type. Originally `hardware` could only be a string or null, but then after this step, it could also be `SpaceHardware`. Now, it can be `SpaceHardware` from the beginning, which might be useful.